### PR TITLE
remote push issue solved.

### PR DESCRIPTION
**Pull request by masumcse1**
Now  meveo will throw a exception  if github repo is already updated. It will show following message . 

image

### CODE REVIEW

File: meveo-admin-ejbs/src/main/java/org/meveo/service/git/GitClient.java [hunk 1]
```diff
@@ -412,17 +414,36 @@ public void push(GitRepository gitRepository, String username, String password)
...
             final PushCommand push = git.push();
             if(gitRepository.getRemoteOrigin().startsWith("http")) {
-                CredentialsProvider usernamePasswordCredentialsProvider = GitHelper.getCredentialsProvider(gitRepository, username, password, user);
-                push.setCredentialsProvider(usernamePasswordCredentialsProvider).call();
+		        CredentialsProvider usernamePasswordCredentialsProvider = GitHelper.getCredentialsProvider(gitRepository, username, password, user);
+		        Iterable<PushResult> pushResults = push.setCredentialsProvider(usernamePasswordCredentialsProvider).call();
+		        for (PushResult pushresult : pushResults) {
+		            for(RemoteRefUpdate remoteRef : pushresult.getRemoteUpdates()) {
+		                if(remoteRef.getStatus() != RemoteRefUpdate.Status.OK && remoteRef.getStatus() != RemoteRefUpdate.Status.UP_TO_DATE) {
+		                    if(remoteRef.getStatus() == RemoteRefUpdate.Status.REJECTED_NONFASTFORWARD) {
```
Comment (smichea) [inline meveo-admin-ejbs/src/main/java/org/meveo/service/git/GitClient.java]: you should throw an exception in any case the status is not OK or UP_TO_DATE, not only in the case it is REJECTED_NONFASTFORWARD

**Review by smichea — COMMENTED** [review]
throw exception for each status that is not OK or UP_TO_DATE

**Comment by masumcse1** [comment]
updated.

**Comment by masumcse1** [comment]
updated as per your feedback.

**Review by smichea — APPROVED** [review]

**Review by smichea — APPROVED** [review]

**Pull request merged**